qt/trafficmodeinitdlg.cpp

```diff
@@ -0,0 +1,64 @@
+#include "trafficmodeinitdlg.h"
+#include "ui_trafficmodeinitdlg.h"
+
+#include "platform/settings.hpp"
+
+#include <QFileDialog>
+
+namespace
+{
+string const kDataFilePath = "LastTrafficDataFilePath";
+string const kSampleFilePath = "LastTrafficSampleFilePath";
+}  // namespace
+
+TrafficModeInitDlg::TrafficModeInitDlg(QWidget *parent) :
+  QDialog(parent),
+  ui(new Ui::TrafficModeInitDlg)
+{
+  ui->setupUi(this);
+
+  string lastDataFilePath;
+  string lastSampleFilePath;
+  if (settings::Get(kDataFilePath, lastDataFilePath) &&
```

Comment:
I suggest you to separate retrieval of last data file path and last sample file path.